python/benchmark/test_gen_data.py

```diff
@@ -103,3 +103,50 @@ def test_make_low_rank_matrix(dtype: str) -> None:
 
         _, s, _ = svd(X)
         assert sum(s) - 5 < 0.1, "X rank is not approximately 5"
+
+
+@pytest.mark.parametrize(
```

Comment:
Missed this before.  Check other examples in the repo on how to set up a multi-parametrized test using multiple parametrize statements.  Don't need to enumerate the combinations as it is done automatically.